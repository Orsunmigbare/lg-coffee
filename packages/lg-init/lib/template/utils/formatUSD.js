/* eslint-disable import/prefer-default-export */
const formatUSD = (number) => {
  const string = JSON.stringify(number);
  const { length } = string;
  return JSON.parse(
    `${string.substring(0, length - 2)}.${string.substring(length - 2)}`
  ).toFixed(2);
};

export { formatUSD };
