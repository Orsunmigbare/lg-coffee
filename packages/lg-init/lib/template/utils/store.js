module.exports = {
  URL: process.env.LOCAL_URL || process.env.PUBLIC_URL,
  storeName: 'VisionSpeaks',
  logo: '/logo.png',
  accountOwner: 'Yulissa V',
  products: [
    {
      name: 'Lady In The Void',
      description: 'Holographic Sticker, This thing is bad ass',
      price: '287',
      sku: 'lad_vs_',
      images: [
        '/lady_in_the_void_TRANSP.png',
        '/lady_001.png',
        '/lady_002.png',
      ],
    },
    {
      name: 'Bunns In The Void',
      description: 'Holographic Sticker, This thing is bad ass',
      price: '287',
      sku: 'bun_vs_',
      images: [
        '/bunns_in_the_void_TRANSP.png',
        '/bunns_001.png',
        '/bunns_002.png',
      ],
    },
  ],
  pages: {
    globals: {
      // navigations
      navigation: [
        {
          title: 'Products',
          path: '/',
        },
        {
          title: 'Made To Order',
          path: '/info/contact',
        },
        {
          title: 'About',
          path: '/info/about',
        },
        {
          title: 'Cart',
          path: '/view-cart',
        },
      ],
      footer: [
        {
          title: 'Contact',
          path: 'mailto:support@visionspeaks.art',
        },
        {
          title: 'FAQ',
          path: '/info/faq',
        },
      ],
    },
    homePage: {
      hero: {
        image: '/heatpress.jpeg',
        headline: 'Made To Order',
        subHeadline: 'The highest quality work comes from custom work.',
        buttonName: 'Start Your Order',
        buttonPath: '/products',
        styles: {},
      },
    },
    info: {
      faq: [
        {
          q: 'Do you make custom products?',
          a: 'Yes',
        },
        {
          q: 'How long will it take to recieve my purchase?',
          a: 'Unless otherwise stated, All orders go out via Priority Mail.',
        },
        {
          q: 'What is your policy on refunds?',
          a: 'At this time, we are unable to honor refunds.',
        },
        {
          q:
            "I never received my purchase or something isn't right with my order. What can I do?",
          a: 'Reach out to support@visionspeaks.art.',
        },
      ],
      about: `
        VisionSpeaks is a custom art collective.
      `,
    },
  },
};
