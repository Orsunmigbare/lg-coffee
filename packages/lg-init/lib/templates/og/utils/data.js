exports.data = {
  store: {
    name: 'VisionSpeaks',
    logo: '/logo.png',
    url: '',
    description: 'Custom sticker shop',
    contact: 'youremail@here.com',
    info: {
      faq: [
        {
          q: 'Do you make custom products?',
          a: 'Yes',
        },
        {
          q: 'How long will it take to recieve my purchase?',
          a: 'Unless otherwise stated, All orders go out via Priority Mail.',
        },
        {
          q: 'What is your policy on refunds?',
          a: 'At this time, we are unable to honor refunds.',
        },
        {
          q:
            "I never received my purchase or something isn't right with my order. What can I do?",
          a: 'Reach out to support@visionspeaks.art.',
        },
      ],
    },
    social: {
      facebook: '',
      twitter: '',
      instagram: '',
    },
  },
  theme: {
    text: { color: '#212529' },
    secondaryText: { color: '#6C757D' },
    primaryBackground: { backgroundColor: '#F8F9FA' },
    secondaryBackground: { backgroundColor: '#FFFFFF' },
  },
  products: {
    heading: 'New Arrivials',
    products: [
      {
        name: 'Bunns In The Void',
        description: 'Holographic Sticker, This thing is bad ass',
        price: 287,
        sku: 'bun_vs_',
        images: [
          '/bunns_in_the_void_TRANSP.png',
          '/bunns_001.png',
          '/bunns_002.png',
        ],
        type: 'sticker',
        qty: 40,
      },
      {
        name: 'Lady In The Void',
        description: 'Holographic Sticker, This thing is bad ass',
        price: 287,
        sku: 'lad_vs_',
        images: [
          '/lady_in_the_void_TRANSP.png',
          '/lady_001.png',
          '/lady_002.png',
        ],
        type: 'sticker',
        qty: 40,
      },
    ],
  },
};
