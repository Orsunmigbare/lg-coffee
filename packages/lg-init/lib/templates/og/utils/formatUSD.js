/* eslint-disable import/prefer-default-export */
const formatUSD = (number) => {     
  const { length } = number;
  const str = `${number.substring(0, length - 2)}.${number.substring(length - 2)}`;  
  return parseFloat(str).toFixed(2)
};

export { formatUSD };
