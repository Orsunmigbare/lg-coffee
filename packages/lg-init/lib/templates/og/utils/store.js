exports.store = {
  name: 'Dog-Gone Goods',
  logo: '/logo.png',
  hero: '/hero.png',
  url: '',
  description: 'A dog treat shop',
  about: 'Everyone loves a happy dog!',
  contact: 'youremail@here.com',
  social: {
    facebook: '',
    twitter: '',
    instagram: '',
  },
};
