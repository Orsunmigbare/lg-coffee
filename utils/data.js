exports.data = {
  store: {
    accountOwner: "Yulissa V",
    dateCreated: "08/2020",
    storeName: "VisionSpeaks",
    logo: "/logo.png",
    hero: {
      heroImage: "/heatpress.jpeg",
      heroHeadline: "Made To Order",
      heroSubHeadline: "The highest quality work comes from custom work.",
      heroButton: "Start Your Order",
      heroPath: "/products",
      styles: {},
    },
    nav: {
      links: [
        {
          title: "Products",
          path: "/",
        },
        {
          title: "Made To Order",
          path: "/made-to-order",
        },
        {
          title: "About",
          path: "/about",
        },
      ],
      style: {
        nav: {
          background: "#FAFAFA",
        },
        link: {
          color: "#181818",
        },
      },
    },
    footer: {
      col: [
        {
          title: "Contact",
          link: "/contact",
        },
        {
          title: "Contact",
          link: "/contact",
        },
      ],
    },
  },
  products: [
    {
      imagePath: "/front_page_bunns@2x.png",
      imagePathHovered: "/front_page_bunns_hover@2x.png", // drop image opacity to .6
      productTitle: "Bunns In The Void",
      productDescription: "Holographic Sticker, This thing is bad ass",
      productPrice: "$2.87",
      productSku: "/bun_vs_",
      productBgIsLight: "true",
    },
    {
      imagePath: "/front_page_lady@2x.png",
      imagePathHovered: "/front_page_lady_hover@2x.png",
      productTitle: "Lady In The Void",
      productDescription: "Holographic Sticker, This thing is bad ass",
      productPrice: "$2.87",
      productSku: "/lad_vs_",
      productBgIsLight: false,
    },
  ],
};
