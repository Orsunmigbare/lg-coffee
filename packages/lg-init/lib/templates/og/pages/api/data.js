import { data } from '../../utils/data';

export default function handler(req, res) {
  res.statusCode = 200;
  res.setHeader('Content-Type', 'application/json');
  return res.end(JSON.stringify(data));
}
