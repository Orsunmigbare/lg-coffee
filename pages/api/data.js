import { data } from "../../utils/data";

export default function handler(req, res) {
  res.statusCode = 200;
  res.setHeader("Content-Type", "application/json");

  // fake db call
  setTimeout(() => {
    res.end(JSON.stringify(data));
  }, 1000);
}
