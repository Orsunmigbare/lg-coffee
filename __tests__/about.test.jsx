import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import About from '../pages/info/about';

// needs to fetch data first
import { data } from '../utils/data';

it('renders smoke test without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<About data={data} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders about page unchanged', () => {
  const tree = renderer.create(<About data={data} />).toJSON();
  expect(tree).toMatchSnapshot();
});
