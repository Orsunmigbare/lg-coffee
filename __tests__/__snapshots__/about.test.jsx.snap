// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders about page unchanged 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light sticky-top"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      VisionSpeaks
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key="/"
          href="/"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Products
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/info/made-to-order"
          href="/info/made-to-order"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Made To Order
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/info/about"
          href="/info/about"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          About
        </a>
      </div>
    </div>
  </nav>,
  <div
    className="container"
    style={
      Object {
        "minHeight": "580px",
      }
    }
  >
    <div
      className="row"
    >
      <div
        className="col-lg-6 col-md-8 col-sm-10 col-12 offset-lg-3 offset-md-2 offset-sm-1"
      >
        <h1
          className="text-light mt-4"
        >
          About
        </h1>
        <p
          className="text-light"
        >
          
        VisionSpeaks is a custom art collective.
      
        </p>
      </div>
    </div>
  </div>,
  <div
    className="my-5 container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="d-flex justify-content-center col"
      >
        <span
          className="text-light m-2"
          onClick={[Function]}
          onMouseEnter={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          Contact
        </span>
        <span
          className="text-light m-2"
          onClick={[Function]}
          onMouseEnter={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          FAQ
        </span>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="d-flex justify-content-center col"
      >
        <span
          className="text-light m-2"
          style={
            Object {
              "fontSize": "12px",
            }
          }
        >
          2020
        </span>
      </div>
    </div>
  </div>,
]
`;

exports[`renders homepage unchanged 1`] = `
<div>
  Hello
</div>
`;
